typescript/sdk/src/token/CosmosNativeWarpModule.ts

```diff
@@ -0,0 +1,469 @@
+   *
+   * @param expectedConfig - The configuration for the token router to be updated.
+   * @returns An array of Cosmos transactions that were executed to update the contract, or an error if the update failed.
+   */
+  async update(
+    expectedConfig: HypTokenRouterConfig,
+  ): Promise<AnnotatedCosmJsNativeTransaction[]> {
+    HypTokenRouterConfigSchema.parse(expectedConfig);
+    const actualConfig = await this.read();
+
+    const transactions = [];
+
+    /**
+     * @remark
+     * The order of operations matter
+     * 1. createOwnershipUpdateTxs() must always be LAST because no updates possible after ownership transferred
+     * 2. createRemoteRoutersUpdateTxs() must always be BEFORE createSetDestinationGasUpdateTxs() because gas enumeration depends on domains
+     */
+    transactions.push(
+      ...(await this.createIsmUpdateTxs(actualConfig, expectedConfig)),
+      ...this.createEnrollRemoteRoutersUpdateTxs(actualConfig, expectedConfig),
+      ...this.createUnenrollRemoteRoutersUpdateTxs(
+        actualConfig,
+        expectedConfig,
+      ),
+      ...(await this.createSetDestinationGasUpdateTxs(
+        actualConfig,
+        expectedConfig,
+      )),
+      ...this.createOwnershipUpdateTxs(actualConfig, expectedConfig),
+    );
+
+    return transactions;
+  }
+
+  /**
+   * Create a transaction to update the remote routers for the Warp Route contract.
+   *
+   * @param actualConfig - The on-chain router configuration, including the remoteRouters array.
+   * @param expectedConfig - The expected token router configuration.
+   * @returns A array with a single Ethereum transaction that need to be executed to enroll the routers
+   */
+  createEnrollRemoteRoutersUpdateTxs(
+    actualConfig: HypTokenRouterConfig,
+    expectedConfig: HypTokenRouterConfig,
+  ): AnnotatedCosmJsNativeTransaction[] {
+    const updateTransactions: AnnotatedCosmJsNativeTransaction[] = [];
+    if (!expectedConfig.remoteRouters) {
+      return [];
+    }
+
+    assert(actualConfig.remoteRouters, 'actualRemoteRouters is undefined');
+    assert(expectedConfig.remoteRouters, 'expectedRemoteRouters is undefined');
+
+    const { remoteRouters: actualRemoteRouters } = actualConfig;
+    const { remoteRouters: expectedRemoteRouters } = expectedConfig;
+
+    const routesToEnroll = Object.entries(expectedRemoteRouters)
+      .filter(([domain, expectedRouter]) => {
+        const actualRouter = actualRemoteRouters[domain];
+        // Enroll if router doesn't exist for domain or has different address
+        return (
+          !actualRouter ||
+          !eqAddress(actualRouter.address, expectedRouter.address)
+        );
+      })
+      .map(([domain]) => domain);
+
+    if (routesToEnroll.length === 0) {
+      return updateTransactions;
+    }
+
+    // in cosmos the gas is attached to the remote router. we set
+    // it to zero for now and set the real value later during the
+    // createSetDestinationGasUpdateTxs step
```

Comment:
I think it would be ok to diverge from the evm "flow" for updating the remote routers config if this helps also in reducing the gas needed for updates considering that the underlying config is different on chain.